### PR TITLE
integrate new self-hosted benchmark runner

### LINKED ISSUE #1100: Integrate new benchmark self-hosted runner into CI
devOps have set up a new self-hosted benchmark runner for us. The runner can be targeted with `[self-hosted, Linux, X64, benchmark]`. Need to integrate it into GitHub actions and report back to the team.  

**Acceptance Criteria**
- [x] The benchmarks need to be triggered on-demand (Trigger TBD)
- [x] Only members of the LibertyDSNP org can trigger benchmark run
- [x] The benchmarks must run only on dedicated benchmark runners
- [x] The PR must be blocked from merging until updated weights have been recorded
- [x] Use `make benchmarks` for generating all weights
- [x] The results with updated weights must be committed back to the PR branch
- [x] The PR owner should be notified when the updated weights have been committed

**Tasks:**
- [x] Validate acceptance criteria with BC team
- [x] Collect feedback on the updated CI diagram
- [x] Implemente feedback
- [x] Implement workflow
- [x] Update Frequency CI diagram to add benchmark workflow
- [x] In PR comment, add link to the commit with updated weights.
- [x] Add instructions to Wiki on how to run benchmarks
- [x] Globally, replace hardcoded `git commit` with `stefanzweifel/git-auto-commit-action` action
- [x] Make "Run Benchmarks" check required after merging PR to `main`
- [x] Mark [Previous Benchmarking Flow Diagram (To Be Updated)](https://docs.google.com/drawings/d/1C_CyH5s7tzAlPTEYKClsSkdpSv95t5pHp_jdpLCt7ss/edit) as obsolete 

References:
- [Benchmarking Wiki](https://github.com/LibertyDSNP/frequency/wiki/Benchmarking)
- [Frequency CI diagram](https://docs.google.com/drawings/d/1IcAzZbulRcntYQ5ub-zDyIkqJY1NE0iqClhkg2PE5lo/edit)
- [Previous Benchmarking Flow Diagram (To Be Updated)](https://docs.google.com/drawings/d/1C_CyH5s7tzAlPTEYKClsSkdpSv95t5pHp_jdpLCt7ss/edit)
- #280
- [Manual Approval GH Action](https://github.com/marketplace/actions/manual-workflow-approval)
- [Block PR via commit status](https://docs.github.com/en/rest/commits/statuses)

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to integrate new self-hosted benchmark runner created by devOps team into the current GItHub Actions CI.

Closes #1100

### CODE REVIEW

**Review by wilwade — APPROVED** [review]

File: .github/workflows/verify-pr-commit.yml
Comment (wilwade) [inline .github/workflows/verify-pr-commit.yml:812]: Do we need to set `ignoreForks: "false"`? We don't really want to run benchmarks without it being from a core dev however. Thoughts?

File: .github/workflows/verify-pr-commit.yml
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:748]: If I rebase and force push, will it re-trigger benchmarks? It will it the last commit was `add another feature [run-benchmarks all]` (desired), but will it trigger if that commit is `HEAD~2+` (Undesired)?

I think the answer is that it works as desired and this `$commit_message` is only the last commit message.

**Review by wilwade — COMMENTED** [review]
Few questions, but looks good

**Review by wilwade — APPROVED** [review]
🥇

**Review by saraswatpuneet — APPROVED** [review]

File: .github/workflows/verify-pr-commit.yml
Comment (JoeCap08055) [reply .github/workflows/verify-pr-commit.yml:801]: I have changes to the benchmark script in the `private-graph-main` branch that will make integration with this easier; I'd be happy to merge that file it into this PR if you like.

**Pull request merged**
